app/domain/authentication/installed_authenticators.rb

```diff
@@ -5,59 +5,80 @@ class InstalledAuthenticators
-        .to_h
-    end
+    class << self
+      def authenticators(env, authentication_module: ::Authentication)
+        loaded_authenticators(authentication_module)
+          .select { |cls| valid?(cls) }
+          .map { |cls| [url_for(cls), authenticator_instance(cls, env)] }
+          .to_h
+      end
 
-    def self.login_authenticators(env, authentication_module: ::Authentication)
-      loaded_authenticators(authentication_module)
-        .select { |cls| provides_login?(cls) }
-        .map { |cls| [url_for(cls), authenticator_instance(cls, env)] }
-        .to_h
-    end
+      def login_authenticators(env, authentication_module: ::Authentication)
+        loaded_authenticators(authentication_module)
+          .select { |cls| provides_login?(cls) }
+          .map { |cls| [url_for(cls), authenticator_instance(cls, env)] }
+          .to_h
+      end
 
-    def self.configured_authenticators
-      identifier = Sequel.function(:identifier, :resource_id)
-      kind = Sequel.function(:kind, :resource_id)
+      def configured_authenticators
+        identifier = Sequel.function(:identifier, :resource_id)
+        kind = Sequel.function(:kind, :resource_id)
 
-      Resource
-        .where(identifier.like("#{AUTHN_RESOURCE_PREFIX}%"))
-        .where(kind => "webservice")
-        .select_map(identifier)
-        .map { |id| id.sub %r{^conjur\/}, "" }
-        .push(::Authentication::Common.default_authenticator_name)
-    end
+        Resource
+          .where(identifier.like("#{AUTHN_RESOURCE_PREFIX}%"))
+          .where(kind => "webservice")
+          .select_map(identifier)
+          .map { |id| id[%r{^conjur\/(authn(?:-[^\/]+)?(?:\/[^\/]+)?)$}, 1] }
+          .compact
+          .push(::Authentication::Common.default_authenticator_name)
+      end
 
-    def self.enabled_authenticators(env)
-      (env["CONJUR_AUTHENTICATORS"] || ::Authentication::Common.default_authenticator_name).split(",")
-    end
+      def enabled_authenticators(env)
+        # Enabling via environment overrides enabling via CLI
+        env_enabled_authenticators(env) || db_enabled_authenticators
+      end
 
-    private
+      def enabled_authenticators_str(env)
+        enabled_authenticators(env).join(',')
+      end
 
-    def self.loaded_authenticators(authentication_module)
-      ::Util::Submodules.of(authentication_module)
-        .flat_map { |mod| ::Util::Submodules.of(mod) }
-    end
+      private
 
-    def self.authenticator_instance(cls, env)
-      pass_env = ::Authentication::AuthenticatorClass.new(cls).requires_env_arg?
-      pass_env ? cls.new(env: env) : cls.new
-    end
+      def env_enabled_authenticators(env)
+        authenticators = env["CONJUR_AUTHENTICATORS"]
+        authenticators.present? ? authenticators.split(',') : nil
+      end
+      
+      def db_enabled_authenticators
+        # For now, always include 'authn' when enabling authenticators via CLI
```

Comment:
What do you mean by "For now"? what can change in the future that will remove `authn` from the enabled authenticators?

Reply:
Removed the "For now".